### PR TITLE
Store Validations

### PR DESCRIPTION
**Pull request by intelliot**
## High Level Overview of Change

Validations are received over the p2p network via a subscription (using websockets to rippled).

This stores the validations in the database.

Validations include `amendments`, an array of amendment IDs that the validator is voting for. When an amendment has the support of 80% of UNL validators, a two-week waiting period begins. If the amendment still has 80% support throughout this period, the amendment will be enabled (via a pseudo-transaction).

This PR updates the Amendment schema and adds support for getting human-readable amendment names from the rippled source code.

### Context of Change

The bot relies on validations in order to count up the number of unique validations from UNL validators that each ledger receives. If the bot does not see a validation from every UNL validator within x seconds, it reports that there is a problem with the missing validator. If the problem persists for subsequent ledgers, the bot should alert.

x = 10. This may be made configurable.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking change that only restructures code)

## Before / After

With the current VCR, validations are stored in memory and are difficult to query. They are lost when the bot restarts.

We now persist them in the database and can query them with SQL.

## Test Plan

Run the bot with `yarn start`. Observe the contents of the `validations` table.

There seems to be a race condition causing duplicate validations to handled multiple times. We should fix this, but it doesn't seem to cause harm; I don't see duplicate data appearing in the database, perhaps thanks to the composite primary key uniqueness constraint.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -43,9 +46,72 @@ createConnection().then(async connection => {
       console.warn(`[${data.validation_public_key_base58}] Ignoring stale manifest # ${data.Sequence}, already had manifest # ${validator.seq} !!`);
     }
   };
-  network.onValidationReceived = (validation) => {
-    // const truncated_hash = validation.ledger_hash.slice(0, 3) + '…' + validation.ledger_hash.slice(-3);
-    // console.log(`Received validation from ${validation.validation_public_key} for ledger ${validation.ledger_index} (${truncated_hash})`);
+  network.onValidationReceived = async (validationMessage) => {
```
Comment (hbergren) [inline src/index.ts]: Is it worth breaking up this function a bit with helper functions?

Maybe:
- Handle validationMessage amendments
- Build validation object
- Save validation

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "moment": "^2.24.0",
+    "node-fetch": "^2.6.0",
```
[the remark this answers is not in the document]
Comment (intelliot) [reply package.json]: Yes

File: src/amendments.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+const cachedAmendmentIDs = {}
+
+async function fetchAmendmentNames() {
+  return fetch('https://raw.githubusercontent.com/ripple/rippled/master/src/ripple/protocol/impl/Feature.cpp')
+    .then(res => res.text())
+    .then(text => {
+      const amendmentNames = []
+      text.split('\n').forEach((line: string) => {
+        const name = (line.match(/^\s*\/?\/?\s*"(.+)",.*$/));
```
[the remark this answers is not in the document]
Comment (intelliot) [reply src/amendments.ts]: Match lines that start with optional whitespace, followed by two optional slashes, followed by optional whitespace, a double quote, any characters*, a double quote, a comma, and ending with any characters.

*Capture the characters between the double quotes, which form the returned `name`

File: src/index.ts (hunk 1 above)
Comment (intelliot) [reply src/index.ts]: https://github.com/xpring-eng/validation-count-reporter/issues/8

**Pull request merged**